app/src/main/java/io/dolby/rtsviewer/ui/streaming/StreamingViewModel.kt

```diff
@@ -430,8 +430,7 @@ class StreamingViewModel @Inject constructor(
         streamQualityType?.let { qualityType ->
             _uiState.update {
                 it.copy(
-                    selectedStreamQualityType = qualityType,
-                    pendingSelectedStreamQualityType = null
+                    selectedStreamQualityType = qualityType
```

Comment:
Don't call playVideo twice in a row, which happens when you null out pendingSelectedStreamQuality, which is being observed